actors/common/src/main/java/org/sunbird/learner/actors/PageManagementActor.java

```diff
@@ -551,33 +554,58 @@ private void updatePageDataCacheHandler(Response response, Map<String, Object> p
     ProjectLogger.log(
         "PageManagementActor:getContentData: Page assemble final search query: " + queryRequestBody,
         LoggerEnum.INFO.name());
-
-    Future<Map<String, Object>> result =
-        ContentSearchUtil.searchContent(urlQueryString, queryRequestBody, headers);
-
-    return result.map(
-        new Mapper<Map<String, Object>, Map<String, Object>>() {
-          @Override
-          public Map<String, Object> apply(Map<String, Object> result) {
-            ProjectLogger.log(
-                "PageManagementActor:getContentData:apply: result = ", LoggerEnum.INFO.name());
-            if (MapUtils.isNotEmpty(result)) {
-              section.putAll(result);
-              section.remove(JsonKey.PARAMS);
-              Map<String, Object> tempMap = (Map<String, Object>) result.get(JsonKey.PARAMS);
-              section.put(JsonKey.RES_MSG_ID, tempMap.get(JsonKey.RES_MSG_ID));
-              section.put(JsonKey.API_ID, tempMap.get(JsonKey.API_ID));
-              section.put(JsonKey.GROUP, group);
-              section.put(JsonKey.INDEX, index);
-              removeUnwantedData(section, "getPageData");
+    Future<Map<String, Object>> result = null;
+    String dataSource = (String) section.get(JsonKey.DATA_SOURCE);
+    if (StringUtils.isEmpty(dataSource) || JsonKey.CONTENT.equalsIgnoreCase(dataSource)) {
+      result = ContentSearchUtil.searchContent(urlQueryString, queryRequestBody, headers);
+      return result.map(
+          new Mapper<Map<String, Object>, Map<String, Object>>() {
+            @Override
+            public Map<String, Object> apply(Map<String, Object> result) {
               ProjectLogger.log(
-                  "PageManagementActor:getContentData:apply: section = " + section,
-                  LoggerEnum.DEBUG.name());
+                  "PageManagementActor:getContentData:apply: result = ", LoggerEnum.INFO.name());
+              if (MapUtils.isNotEmpty(result)) {
+                section.putAll(result);
+                section.remove(JsonKey.PARAMS);
```

Comment:
section.remove(jsonkey.Params)
section.put  for Group, index 

are repeated in if/else both sections - so